client-src/index.js

```diff
@@ -118,9 +118,15 @@ self.addEventListener("beforeunload", () => {
 const trustedTypesPolicyName =
   typeof options.overlay === "object" && options.overlay.trustedTypesPolicyName;
 
-const overlay = createOverlay({
-  trustedTypesPolicyName,
-});
+const overlay = options.overlay
```

Comment:
@malcolm-kee  this is not sufficient. You are only checking if `options.overlay` is truthy. All options need to be passed into the `createOverlay()` function. The `overlay` options are `errors: boolean` & `warnings: boolean`. 

Then, the `show()` function in `overlay.js` needs to check if the `type` is "warning" or "error" and compare the passed options.

Reply:
Yeah, we have before https://github.com/webpack/webpack-dev-server/blob/v4.11.1/client-src/index.js#L253

Reply:
@crobinson42 good catch!

I just pushed a fix so that runtime error check the options as well.

For build error/warning, it's already handled:

- (errors) https://github.com/webpack/webpack-dev-server/blob/master/client-src/index.js#L297
- (warnings) https://github.com/webpack/webpack-dev-server/blob/master/client-src/index.js#L265

Reply:
Testa are failed, can you look what is wrong (no rush), I think today after rest we will fix all issues